### PR TITLE
ATS-886 /  ATS-907: Automation tests: light-weight management of rich media files (IPTC metadata extract)

### PR DESCRIPTION
**Pull request by aymanthefirst**
Added unit tests in alfresco-transform-tika\alfresco-transform-tika-boot\src\test\java\org\alfresco\transformer\TikaMetadataExtractsIT.java and their corresponding quick files, to test the IPTC metadata extractions as listed by file type in alfresco-transform-tika/alfresco-transform-tika/src/main/resources/tika_engine_config.json#L532-L548

### CODE REVIEW

File: alfresco-transform-tika/alfresco-transform-tika-boot/src/test/java/org/alfresco/transformer/TikaMetadataExtractsIT.java [hunk 1]
```diff
@@ -29,43 +29,7 @@
...
-import static org.alfresco.transformer.util.MimetypeMap.MIMETYPE_VIDEO_3GP2;
-import static org.alfresco.transformer.util.MimetypeMap.MIMETYPE_VIDEO_FLV;
-import static org.alfresco.transformer.util.MimetypeMap.MIMETYPE_VIDEO_MP4;
-import static org.alfresco.transformer.util.MimetypeMap.MIMETYPE_VIDEO_QUICKTIME;
-import static org.alfresco.transformer.util.MimetypeMap.MIMETYPE_VISIO;
-import static org.alfresco.transformer.util.MimetypeMap.MIMETYPE_VORBIS;
-import static org.alfresco.transformer.util.MimetypeMap.MIMETYPE_WORD;
-import static org.alfresco.transformer.util.MimetypeMap.MIMETYPE_XML;
-import static org.alfresco.transformer.util.MimetypeMap.MIMETYPE_ZIP;
+import static org.alfresco.transformer.util.MimetypeMap.*;
```
Comment (PunchedChimera) [inline alfresco-transform-tika/alfresco-transform-tika-boot/src/test/java/org/alfresco/transformer/TikaMetadataExtractsIT.java:32]: Please revert and add specific variables that are being imported.

File: alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.m4a_metadata.json [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-  "{http://www.alfresco.org/model/content/1.0}description" : "The quick brown fox jumps over the lazy dog - About a dog and a fox (Hauskaz)",
-  "{http://www.alfresco.org/model/audio/1.0}releaseDate" : 1230768000000,
-  "{http://www.alfresco.org/model/audio/1.0}channelType" : "Stereo",
-  "{http://www.alfresco.org/model/content/1.0}created" : 1230768000000,
-  "{http://www.alfresco.org/model/audio/1.0}sampleRate" : "44100",
-  "{http://www.alfresco.org/model/content/1.0}author" : "Hauskaz",
-  "{http://www.alfresco.org/model/audio/1.0}album" : "About a dog and a fox",
-  "{http://www.alfresco.org/model/content/1.0}title" : "The quick brown fox jumps over the lazy dog"
+{
+  "{http://www.alfresco.org/model/audio/1.0}artist" : "Hauskaz",
```
Comment (PunchedChimera) [inline alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.m4a_metadata.json:2]: Please revert this file. m4a is not handled by IPTC extraction so should not change.

File: alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.mp3_metadata.json [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
-  "{http://www.alfresco.org/model/content/1.0}description" : "The quick brown fox jumps over the lazy dog - About a dog and a fox (Hauskaz)",
-  "{http://www.alfresco.org/model/audio/1.0}releaseDate" : 1230768000000,
-  "{http://www.alfresco.org/model/audio/1.0}channelType" : "Stereo",
-  "{http://www.alfresco.org/model/content/1.0}created" : 1230768000000,
-  "{http://www.alfresco.org/model/audio/1.0}sampleRate" : "44100",
-  "{http://www.alfresco.org/model/content/1.0}author" : "Hauskaz",
-  "{http://www.alfresco.org/model/audio/1.0}album" : "About a dog and a fox",
-  "{http://www.alfresco.org/model/content/1.0}title" : "The quick brown fox jumps over the lazy dog"
+{
+  "{http://www.alfresco.org/model/audio/1.0}artist" : "Hauskaz",
```
Comment (PunchedChimera) [inline alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.mp3_metadata.json:2]: Please revert this file. mp3 is not handled by IPTC extraction so should not change.

File: alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.odf_metadata.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
-{
-  "{http://www.alfresco.org/model/content/1.0}description" : "Gym class featuring a brown fox and lazy dog",
-  "{http://www.alfresco.org/model/content/1.0}created" : 1138362922000,
-  "{http://www.alfresco.org/model/content/1.0}author" : null,
-  "{http://www.alfresco.org/model/content/1.0}title" : "The quick brown fox jumps over the lazy dog"
+{
+  "{http://www.alfresco.org/model/content/1.0}description" : "Gym class featuring a brown fox and lazy dog",
+  "{http://www.alfresco.org/model/content/1.0}created" : 1138370122000,
```
Comment (PunchedChimera) [inline alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.odf_metadata.json:3]: Please revert this file. odf is not handled by IPTC extraction so should not change.

File: alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.odg_metadata.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
-{
-  "{http://www.alfresco.org/model/content/1.0}description" : "Gym class featuring a brown fox and lazy dog",
-  "{http://www.alfresco.org/model/content/1.0}created" : 1138362371000,
-  "{http://www.alfresco.org/model/content/1.0}author" : "Derek Hulley",
-  "{http://www.alfresco.org/model/content/1.0}title" : "The quick brown fox jumps over the lazy dog"
+{
+  "{http://www.alfresco.org/model/content/1.0}description" : "Gym class featuring a brown fox and lazy dog",
+  "{http://www.alfresco.org/model/content/1.0}created" : 1138369571000,
```
Comment (PunchedChimera) [inline alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.odg_metadata.json:3]: Please revert this file. odg is not handled by IPTC extraction so should not change.

File: alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.ods_metadata.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
-{
-  "{http://www.alfresco.org/model/content/1.0}description" : "Gym class featuring a brown fox and lazy dog",
-  "{http://www.alfresco.org/model/content/1.0}created" : 845336008000,
-  "{http://www.alfresco.org/model/content/1.0}author" : "Derek Hulley",
-  "{http://www.alfresco.org/model/content/1.0}title" : "The quick brown fox jumps over the lazy dog"
+{
+  "{http://www.alfresco.org/model/content/1.0}description" : "Gym class featuring a brown fox and lazy dog",
+  "{http://www.alfresco.org/model/content/1.0}created" : 845343208000,
```
Comment (PunchedChimera) [inline alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.ods_metadata.json:3]: Please revert this file. ods is not handled by IPTC extraction so should not change.

File: alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.odt_metadata.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
-{
-  "{http://www.alfresco.org/model/content/1.0}description" : "Gym class featuring a brown fox and lazy dog",
-  "{http://www.alfresco.org/model/content/1.0}created" : 1126049640000,
-  "{http://www.alfresco.org/model/content/1.0}author" : "Jesper Steen Møller",
-  "{http://www.alfresco.org/model/content/1.0}title" : "The quick brown fox jumps over the lazy dog"
+{
+  "{http://www.alfresco.org/model/content/1.0}description" : "Gym class featuring a brown fox and lazy dog",
+  "{http://www.alfresco.org/model/content/1.0}created" : 1126060440000,
```
Comment (PunchedChimera) [inline alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.odt_metadata.json:3]: Please revert this file. odt is not handled by IPTC extraction so should not change.

File: alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.ogg_metadata.json [hunk 8]
```diff
@@ -1,13 +1,13 @@
...
-  "{http://www.alfresco.org/model/content/1.0}description" : "The quick brown fox jumps over the lazy dog - About a dog and a fox (Hauskaz)",
-  "{http://www.alfresco.org/model/audio/1.0}releaseDate" : 1230768000000,
-  "{http://www.alfresco.org/model/audio/1.0}channelType" : "Stereo",
-  "{http://www.alfresco.org/model/content/1.0}created" : 1230768000000,
-  "{http://www.alfresco.org/model/audio/1.0}sampleRate" : "44100",
-  "{http://www.alfresco.org/model/content/1.0}author" : "Hauskaz",
-  "{http://www.alfresco.org/model/audio/1.0}album" : "About a dog and a fox",
-  "{http://www.alfresco.org/model/content/1.0}title" : "The quick brown fox jumps over the lazy dog"
+{
+  "{http://www.alfresco.org/model/audio/1.0}artist" : "Hauskaz",
```
Comment (PunchedChimera) [inline alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.ogg_metadata.json:2]: Please revert this file. ogg is not handled by IPTC extraction so should not change.

File: alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.otg_metadata.json [hunk 9]
```diff
@@ -1,6 +1,6 @@
-{
-  "{http://www.alfresco.org/model/content/1.0}description" : "Gym class featuring a brown fox and lazy dog",
-  "{http://www.alfresco.org/model/content/1.0}created" : 1138362371000,
-  "{http://www.alfresco.org/model/content/1.0}author" : "Derek Hulley",
-  "{http://www.alfresco.org/model/content/1.0}title" : "The quick brown fox jumps over the lazy dog"
+{
+  "{http://www.alfresco.org/model/content/1.0}description" : "Gym class featuring a brown fox and lazy dog",
+  "{http://www.alfresco.org/model/content/1.0}created" : 1138369571000,
```
Comment (PunchedChimera) [inline alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.otg_metadata.json:3]: Please revert this file. otg is not handled by IPTC extraction so should not change.

File: alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.otp_metadata.json [hunk 10]
```diff
@@ -1,6 +1,6 @@
-{
-  "{http://www.alfresco.org/model/content/1.0}description" : "Gym class featuring a brown fox and lazy dog",
-  "{http://www.alfresco.org/model/content/1.0}created" : 1179313846000,
-  "{http://www.alfresco.org/model/content/1.0}author" : "Derek Hulley",
-  "{http://www.alfresco.org/model/content/1.0}title" : "The quick brown fox jumps over the lazy dog"
+{
+  "{http://www.alfresco.org/model/content/1.0}description" : "Gym class featuring a brown fox and lazy dog",
+  "{http://www.alfresco.org/model/content/1.0}created" : 1179324646000,
```
Comment (PunchedChimera) [inline alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.otp_metadata.json:3]: Please revert this file. otp is not handled by IPTC extraction so should not change.

File: alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.ots_metadata.json [hunk 11]
```diff
@@ -1,6 +1,6 @@
-{
-  "{http://www.alfresco.org/model/content/1.0}description" : "Gym class featuring a brown fox and lazy dog",
-  "{http://www.alfresco.org/model/content/1.0}created" : 845336008000,
-  "{http://www.alfresco.org/model/content/1.0}author" : "Derek Hulley",
-  "{http://www.alfresco.org/model/content/1.0}title" : "The quick brown fox jumps over the lazy dog"
+{
+  "{http://www.alfresco.org/model/content/1.0}description" : "Gym class featuring a brown fox and lazy dog",
+  "{http://www.alfresco.org/model/content/1.0}created" : 845343208000,
```
Comment (PunchedChimera) [inline alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.ots_metadata.json:3]: Please revert this file. ots is not handled by IPTC extraction so should not change.

File: alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.ott_metadata.json [hunk 12]
```diff
@@ -1,6 +1,6 @@
-{
-  "{http://www.alfresco.org/model/content/1.0}description" : "Gym class featuring a brown fox and lazy dog",
-  "{http://www.alfresco.org/model/content/1.0}created" : 1126049640000,
-  "{http://www.alfresco.org/model/content/1.0}author" : "Jesper Steen Møller",
-  "{http://www.alfresco.org/model/content/1.0}title" : "The quick brown fox jumps over the lazy dog"
+{
+  "{http://www.alfresco.org/model/content/1.0}description" : "Gym class featuring a brown fox and lazy dog",
+  "{http://www.alfresco.org/model/content/1.0}created" : 1126060440000,
```
Comment (PunchedChimera) [inline alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.ott_metadata.json:3]: Please revert this file. ott is not handled by IPTC extraction so should not change.

File: alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.sxw_metadata.json [hunk 13]
```diff
@@ -1,6 +1,6 @@
-{
-  "{http://www.alfresco.org/model/content/1.0}description" : "Gym class featuring a brown fox and lazy dog",
-  "{http://www.alfresco.org/model/content/1.0}created" : 1126049640000,
-  "{http://www.alfresco.org/model/content/1.0}author" : "Jesper Steen Møller",
-  "{http://www.alfresco.org/model/content/1.0}title" : "The quick brown fox jumps over the lazy dog"
+{
+  "{http://www.alfresco.org/model/content/1.0}description" : "Gym class featuring a brown fox and lazy dog",
+  "{http://www.alfresco.org/model/content/1.0}created" : 1126060440000,
```
Comment (PunchedChimera) [inline alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.sxw_metadata.json:3]: Please revert this file. sxw is not handled by IPTC extraction so should not change.

File: alfresco-transformer-base/src/main/java/org/alfresco/transformer/util/MimetypeMap.java [hunk 14]
```diff
@@ -105,6 +105,7 @@
     String MIMETYPE_IMAGE_RAW_RW2 = "image/x-raw-panasonic";
     String MIMETYPE_IMAGE_RAW_RWL = "image/x-raw-leica";
     String MIMETYPE_IMAGE_RAW_R3D = "image/x-raw-red";
+    String MIMETYPE_IMAGE_RAW_webp = "image/webp";
```
Comment (PunchedChimera) [inline alfresco-transformer-base/src/main/java/org/alfresco/transformer/util/MimetypeMap.java:108]: Consistency
```suggestion
    String MIMETYPE_IMAGE_RAW_WEBP = "image/webp";
```

**Review by PunchedChimera — CHANGES_REQUESTED** [review]
A lot of files have been modified that shouldn't have. Please revert.

File: alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.tiff_metadata.json [hunk 15]
```diff
@@ -1,16 +1,11 @@
 {
   "{http://www.alfresco.org/model/exif/1.0}pixelYDimension" : "584",
   "{http://www.alfresco.org/model/content/1.0}description" : "Gym class featuring a brown fox and lazy dog",
-  "{http://purl.org/dc/elements/1.1/}title" : "The quick brown fox jumps over the lazy dog",
```
Comment (PunchedChimera) [inline alfresco-transform-tika/alfresco-transform-tika-boot/src/test/resources/quick.tiff_metadata.json:4]: This doesn't look right? Why have you removed the ITPC metadata?

**Comment by aymanthefirst** [comment]
This PR can now be closed because it has been broken down into two PRs, of which the [first](https://github.com/Alfresco/alfresco-transform-core/pull/398) is merged.

**Pull request closed without merging**